docs/vsphere_ci/build.json

```diff
@@ -1,82 +1,82 @@
-         "insecure_connection":"true",
-         "iso_paths":[
-            "{{user `os_iso_path`}}",
-            "[WorkloadDatastore] iso/windows.iso"
-         ],
-         "network_adapters":[
-            {
-               "network":"{{user `vsphere-network`}}",
-               "network_card":"vmxnet3"
-            }
-         ],
-         "password":"{{user `vsphere-password`}}",
-         "storage":[
-            {
-               "disk_size":"{{user `vm-disk-size`}}",
-               "disk_thin_provisioned":true
-            }
-         ],
-         "type":"vsphere-iso",
-         "username":"{{user `vsphere-user`}}",
-         "vcenter_server":"{{user `vsphere-server`}}",
-         "vm_name":"{{user `vm-name`}}",
-         "ssh_password":"{{user `winadmin-password`}}",
-         "ssh_username":"Administrator",
-         "ssh_timeout":"50m"
-      }
-   ],
-   "provisioners":[
-      {
-	"pause_before": "120s",
-	"inline":["ipconfig"],
-        "type":"powershell",
-	"elevated_user": "Administrator",
-        "elevated_password": "{{user `winadmin-password`}}",
-	"max_retries":"10"
-      }
-   ],
-   "sensitive-variables":[
-      "vsphere_password",
-      "winadmin_password"
-   ],
-   "variables":{
-      "os_iso_path":"[WorkloadDatastore] iso/1909_Jan2021.iso",
-      "vm-cpu-num":"2",
-      "vm-disk-size":"128000",
-      "vm-mem-size":"4096",
-      "vm-name":"Win1909-Template-Base",
-      "vsphere-cluster":"Cluster-1",
-      "vsphere-datacenter":"SDDC-Datacenter",
-      "vsphere-datastore":"WorkloadDatastore",
-      "vsphere-folder":"windows-eval-iso",
-      "vsphere-network":"dev-segment",
-      "vsphere-password":"Your_password",
-      "vsphere-server":"vcenter_url",
-      "vsphere-user":"Your_username",
-      "winadmin-password":"Some random password"
-   }
+  "variables": {
+    "os-iso-path": "[WorkloadDatastore] windows-iso-images/winsrv2004_sep_2020_x64_dvd.iso",
+    "vmtools-iso-path": "[WorkloadDatastore] windows-iso-images/vmtools-v11333-windows.iso",
+    "vm-template-folder": "windows-golden-images",
+    "vm-template-name": "vm-winsrv-2004-golden-image-template",
+    "vm-elevated-password": "Some random password",
+    "vsphere-cluster": "Cluster-1",
+    "vsphere-datacenter": "SDDC-Datacenter",
+    "vsphere-datastore": "WorkloadDatastore",
+    "vsphere-server": "vcenter_url",
+    "vsphere-user": "vcenter_username",
+    "vsphere-password": "vcenter_password"
+  },
+  "sensitive-variables": [
+    "vsphere-password",
+    "vm-elevated-password"
+  ],
+  "provisioners": [
+    {
+      "pause_before": "120s",
+      "inline": [
+        "ipconfig"
+      ],
+      "type": "powershell",
+      "elevated_user": "Administrator",
+      "elevated_password": "{{user `vm-elevated-password`}}",
+      "max_retries": "10"
+    }
+  ],
+  "builders": [
+    {
+      "CPUs": "2",
+      "RAM": "4096",
+      "RAM_reserve_all": true,
+      "communicator": "ssh",
+      "convert_to_template": "true",
```

Comment:
You had mentioned that the resulting entity is not a template but a VM. So why is this set?

Reply:
Didn't went well last time I tuned this one, still will take a closer look.

Reply:
Sorry I don't follow. If this did not get well before, why did you create the current golden image with this option?

Reply:
Let me add more color to that statement.

As per current [documentation](https://github.com/openshift/windows-machine-config-operator/blob/master/docs/vsphere_ci/README.md#post-packer-build), the process creates a template (given this flag is set to `true`) and need to be manually converted to a VM in the vCenter.

Had mixed feelings about that too, so explored to following two scenarios:
1- disabling this flag (`convert_to_template: false`) and directly producing a VM, to be used as the golden image.
2- leaving this flag enabled, and using the template, as is in the [template field](https://github.com/openshift/windows-machine-config-operator/blob/820507d6ad0bcfb5d8cd6bad7686a8bbfa087612/test/e2e/providers/vsphere/vsphere.go#L57) of the e2e test, where the [VSphereMachineProviderSpec](https://github.com/openshift/windows-machine-config-operator/blob/595f2828d9aea5c6aefb1a168d4264ed78c83d30/vendor/github.com/openshift/machine-api-operator/pkg/apis/vsphereprovider/v1beta1/vsphereproviderconfig_types.go#L27) is configured.

Back then, none of the above seemed to generate a valid golden image. Since, there were a few inconsistencies in the process will revisit them.